src/axom/core/Path.cpp

```diff
@@ -16,8 +16,15 @@ namespace axom
 {
 Path::Path(const std::string& path, const char delim) : m_delim(delim)
 {
+  size_t first_position = 0;  // position of first non-delimiter char in path
+  if(!path.empty() && path[0] == m_delim)
```

Comment:
Suggestion: Consider using `axom::utilities::string::startsWith()` for this test

https://github.com/LLNL/axom/blob/cd5866a630ada425e9f0f0de0c54c4708543f48c/src/axom/core/utilities/StringUtilities.hpp#L55-L64